lib/streamlit/elements/arrow_altair.py

```diff
@@ -59,14 +68,34 @@ class ChartType(Enum):
     LINE = "line"
 
 
+COLOR_LEGEND_SETTINGS = dict(titlePadding=0, offset=10, orient="bottom")
+
+# User-readable names to give the index and melted columns.
+SEPARATED_INDEX_COLUMN_TITLE = "index"
+MELTED_Y_COLUMN_TITLE = "value"
+MELTED_COLOR_COLUMN_TITLE = "color"
+
+# Crazy internal (non-user-visible) names for the index and melted columns, in order to
+# avoid collision with existing column names.
+PROTECTION_SUFFIX = "--p5bJXXpQgvPz6yvQMFiy"
```

Comment:
Was there a particular reason why this suffix was chosen? It's pretty crazy 😮

Reply:
Just trying to avoid conflicts.

I used a random number generator, not just because it's [guaranteed to be random](https://xkcd.com/221/) but also because it makes it even less likely to lead to a conflict than something that's human-readable (like "--streamlit-fake-field" or something).

Reply:
Added comment.